### ISSUE TITLE
CryptographyDeprecationWarning on module x509_certificate_info

### ISSUE THREAD

**Issue by spartako** [ansible-collections/community.crypto#726]
##### SUMMARY
I'm getting deprecation warnings when using the x509_certificate_info module. E.g:
```
.ansible/collections/ansible_collections/community/crypto/plugins/module_utils/crypto/module_backends/certificate_info.py:325: CryptographyDeprecationWarning: Properties that return a naïve datetime object have been deprecated. Please switch to not_valid_before_utc.
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.crypto.x509_certificate_info

##### ANSIBLE VERSION

```
ansible [core 2.16.5]
  config file = <redacted>
  configured module search path = <redacted>
  ansible python module location = /usr/local/lib/python3.11/dist-packages/ansible
  ansible collection location = <redacted>
  executable location = /usr/local/bin/ansible
  python version = 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] (/usr/bin/python3)
  jinja version = 3.1.3
  libyaml = True
```

##### COLLECTION VERSION
```
Collection       Version
---------------- -------
community.crypto 2.18.0 
```

##### CONFIGURATION
no special config

##### OS / ENVIRONMENT
irrelevant

##### STEPS TO REPRODUCE

just run this in a role

```yaml
- debug:
    var: >-
      '-----BEGIN CERTIFICATE-----
      MIIEijCCA3KgAwIBAgIRAP4dE5JhiucMEGWQKlXQQa0wDQYJKoZIhvcNAQELBQAw
      RjELMAkGA1UEBhMCVVMxIjAgBgNVBAoTGUdvb2dsZSBUcnVzdCBTZXJ2aWNlcyBM
      TEMxEzARBgNVBAMTCkdUUyBDQSAxQzMwHhcNMjQwMzA0MDcxOTA3WhcNMjQwNTI3
      MDcxOTA2WjAZMRcwFQYDVQQDEw53d3cuZ29vZ2xlLmNvbTBZMBMGByqGSM49AgEG
      CCqGSM49AwEHA0IABMMDSQ2mMkIXUjnAeS3yfWjwitq5YVFHILRKzj3K431rP+/j
      FAXZijbgZP+mm7nfJJsy+TXaYgO01q5IdDVDqBWjggJpMIICZTAOBgNVHQ8BAf8E
      BAMCB4AwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAdBgNVHQ4E
      FgQUV/1e5rYZ0yR7pC592IZRcCllWLowHwYDVR0jBBgwFoAUinR/r4XN7pXNPZzQ
      4kYU83E1HScwagYIKwYBBQUHAQEEXjBcMCcGCCsGAQUFBzABhhtodHRwOi8vb2Nz
      cC5wa2kuZ29vZy9ndHMxYzMwMQYIKwYBBQUHMAKGJWh0dHA6Ly9wa2kuZ29vZy9y
      ZXBvL2NlcnRzL2d0czFjMy5kZXIwGQYDVR0RBBIwEIIOd3d3Lmdvb2dsZS5jb20w
      IQYDVR0gBBowGDAIBgZngQwBAgEwDAYKKwYBBAHWeQIFAzA8BgNVHR8ENTAzMDGg
      L6AthitodHRwOi8vY3Jscy5wa2kuZ29vZy9ndHMxYzMvemRBVHQwRXhfRmsuY3Js
      MIIBBgYKKwYBBAHWeQIEAgSB9wSB9ADyAHcASLDja9qmRzQP5WoC+p0w6xxSActW
      3SyB2bu/qznYhHMAAAGOCIuRPAAABAMASDBGAiEAn1g5+kQpvQpi3+hvUTOSufLt
      kTAJGJhKAjtjqx+N7/0CIQCGvjaQJIDJtjzgAZCnj4TpzNOBFLWRqGjR+IfDXRzy
      mgB3AO7N0GTV2xrOxVy3nbTNE6Iyh0Z8vOzew1FIWUZxH7WbAAABjgiLkRcAAAQD
      AEgwRgIhANSmnt2rCCaon3Tlu4rKuxixvrBxMM2VuHeFP9JpvWa/AiEAoPH+GHpK
      WDN4pvPqRvYweky6Ud6mH/RD0x3uiV/8p0owDQYJKoZIhvcNAQELBQADggEBAL7C
      TTtTWrnwz16zmWgr4LDCacIEPO7tiWikxijBneH5odCyoKYfOHmJeMRLTCELAo9e
      EUT00UBv+C+IuTQYqznd26c7FaIfJMa7t+sCFid+QDTISyAbgzgUE/7i9iYBwteD
      PzcEENWXO/ctzGxHqNwA2XBZNNyIhpNQvxHSZ9S36nsOk4fiTnirUMOrXZKfp60j
      qbcyShje65KcwHccLZWlETXGI8uhYD3zkDbRBPXMy0Z1TIhHTKwE+SKSQUBIoRKS
      QqN3IYXEpzXTpoo182hRXwGNc6oCkZumqmtXsVC7oZRU3Kb1A/lO2DG8yIA91Ixy
      EoACZYQWDlFet3okex8=
      -----END CERTIFICATE-----
      ' | community.crypto.x509_certificate_info
```

##### EXPECTED RESULTS
No CryptographyDeprecationWarning

##### ACTUAL RESULTS
you get deprecation warnings coming from the cryptography module
```
.ansible/collections/ansible_collections/community/crypto/plugins/module_utils/crypto/module_backends/certificate_info.py:328: CryptographyDeprecationWarning: Properties that return a naïve datetime object have been deprecated. Please switch to not_valid_after_utc.
  return self.cert.not_valid_after
```

**Comment by felixfontein** [comment]
Thanks for reporting this! I also saw similar warnings today in another app using cryptography. Will take a look this evening / during the next days...

**Comment by felixfontein** [comment]
I've started working on a fix in #727.
